apis/v1alpha2/gateway_types.go

```diff
@@ -198,18 +209,7 @@ type Listener struct {
 	// +optional
 	TLS *GatewayTLSConfig `json:"tls,omitempty"`
 
-	// Routes specifies a schema for associating routes with the
-	// Listener using selectors. A Route is a resource capable of
-	// servicing a request and allows a cluster operator to expose
-	// a cluster resource (i.e. Service) by externally-reachable
-	// URL, load-balance traffic and terminate SSL/TLS.  Typically,
-	// a route is a "HTTPRoute" or "TCPRoute" in group
-	// "gateway.networking.k8s.io", however, an implementation may support
-	// other types of resources.
-	//
-	// The Routes selector MUST select a set of objects that
-	// are compatible with the application protocol specified in
-	// the Protocol field.
+	// Routes specifies which Routes may be attached to this Listener.
```

Comment:
I think the most likely use case of this would be a direct ref from Gateway -> Service, but I think it's best to leave that out of scope for now. I think Routes are a pretty broad and generic category that should be able to fit most things a Gateway would want to route to.